src/Commands/RunRepeater.ts

```diff
@@ -1,16 +1,124 @@
-import { Cert, RequestExecutorOptions } from '../RequestExecutor';
+import {
+  Cert,
+  CertificatesLoader,
+  RequestExecutorOptions
+} from '../RequestExecutor';
 import { ErrorMessageFactory, Helpers, logger } from '../Utils';
 import container from '../container';
 import { DefaultRepeaterServerOptions, RepeaterLauncher } from '../Repeater';
 import { Arguments, Argv, CommandModule } from 'yargs';
 import { captureException } from '@sentry/node';
-import { normalize } from 'node:path';
+import { normalize, basename, extname } from 'node:path';
+import { readFile } from 'node:fs/promises';
+import { createSecureContext } from 'node:tls';
 import process from 'node:process';
+import https from 'node:https';
 
 export class RunRepeater implements CommandModule {
   public readonly command = 'repeater [options]';
   public readonly describe = 'Starts an on-prem agent.';
 
+  public static async isValidCertificate(cert: Cert): Promise<boolean> {
+    const ext = extname(cert.path);
+    if (ext !== '.pfx') {
+      logger.warn('Non-PFX certificates are passed without validation');
+
+      return true;
+    }
+
+    const loadedCert: { cert: Buffer; passphrase?: string } | undefined =
+      await RunRepeater.loadPfxCert(cert);
+
+    if (!loadedCert) {
+      return false;
+    }
+
+    const options = {
+      hostname: cert.hostname,
+      port: cert.port || 443,
+      protocol: 'https:',
+      path: cert.validationPath || '/',
+      method: 'GET',
+      rejectUnauthorized: true,
+      requestCert: true,
+      ca: https.globalAgent.options.ca,
+      agent: new https.Agent({
+        pfx: loadedCert.cert,
+        passphrase: loadedCert.passphrase
+      })
+    };
+
+    try {
+      return await new Promise<boolean>((resolve, reject) => {
+        const req = https.request(options, (res) => {
+          res.on('data', () => {
+            // We are not interested in response data
+            // but need to handle this event not to get crashed.
+          });
+          res.on('end', () => {
+            if (res.statusCode >= 200 && res.statusCode < 400) {
```

Comment:
There is edge case: cert fits in and works, but target responds with statusCode>400. It shouldn't be treated as cert failure.